pkg/kubeapiserver/options/cloudprovider.go

```diff
@@ -32,9 +37,35 @@ func NewCloudProviderOptions() *CloudProviderOptions {
 }
 
 // Validate checks invalid config
-func (s *CloudProviderOptions) Validate() []error {
-	allErrors := []error{}
-	return allErrors
+func (opts *CloudProviderOptions) Validate() []error {
+	var errs []error
+
+	switch {
+	case opts.CloudProvider == "":
+	case opts.CloudProvider == "external":
+		if !utilfeature.DefaultFeatureGate.Enabled(features.DisableCloudProviders) {
+			errs = append(errs, fmt.Errorf("when using --cloud-provider set to '%s', "+
+				"please set DisableCloudProviders feature to true", opts.CloudProvider))
+		}
+		if !utilfeature.DefaultFeatureGate.Enabled(features.DisableKubeletCloudCredentialProviders) {
+			errs = append(errs, fmt.Errorf("when using --cloud-provider set to '%s', "+ //nolint:staticcheck,ineffassign // false positive
+				"please set DisableKubeletCloudCredentialProviders feature to true", opts.CloudProvider))
+		}
+		return nil
```

Comment:
Sorry, late to the party, I missed all the fun and the PR is already merged :cry: 

I think the linter was right: `errs` gets updated, but then isn't used because the code path returns `nil` instead of `errs`.

Reply:
Perhaps add unit tests, too?

Reply:
indeed. Fixing. Unit test sounds like overkill, as everybody of us should have seen this mistake :) But 🤷‍♂️. So a unit test might be a good idea.